src/Repeater/DefaultRepeater.ts

```diff
@@ -0,0 +1,147 @@
+import { logger } from '../Utils';
+import {
+  Repeater,
+  RepeaterEventHandler,
+  RepeaterDeployedEvent,
+  RepeaterEvents
+} from './Repeater';
+import { inject, injectable } from 'tsyringe';
+import io, { Socket } from 'socket.io-client';
+import parser from 'socket.io-msgpack-parser';
+import { hostname } from 'os';
+import { once } from 'events';
+
+export interface DefaultRepeaterOptions {
+  readonly uri: string;
+  readonly token: string;
+}
+
+export const DefaultRepeaterOptions: unique symbol = Symbol(
+  'DefaultRepeaterOptions'
+);
+
+@injectable()
+export class DefaultRepeater implements Repeater {
+  private latestReconnectionError?: Error;
+  private readonly DEFAULT_RECONNECT_TIMES = 3;
```

Comment:
You must preserve the existing behavior:

```suggestion
  private readonly DEFAULT_RECONNECT_TIMES = 20;
```